### PR TITLE
Equal opportunity Classifier

### PR DESCRIPTION
**Pull request by MBrouns**
We already had our FairClassifier which accounted for Demographic Parity with regards to sensitive attributes, but of course no metric is perfect. If we for example use this in a loan approval situation, the demographic parity only looks at loans given, and not at the rate at which loans are paid back. That might result in a lower percentage of qualified people who are given loans in one population than in another. Another way of measuring fairness could be therefore to measure __equal opportunity__ instead. This constraint would boil down to:

$$P(\hat{y}=1 | z=0, y=1)=P(\hat{y}=1 | z=1, y=1)$$

and be turned into a metric as such:

$$\text{equality of opportunity} = \min \left(\frac{P(\hat{y}=1 | z=1, y=1)}{P(\hat{y}=1 | z=0, y=1)}, \frac{P(\hat{y}=1 | z=0, y=1)}{P(\hat{y}=1 | z=1, y=1)}\right)$$****

This PR deprecates the original `FairClassifier` and renames it to `DemographicParityClassifier`. Furthermore, it adds an `equal_opportunity_score` metric and `EqualOpportunityClassifier` model

### CODE REVIEW

File: sklego/linear_model.py [hunk 1]
```diff
@@ -231,4 +189,150 @@ def _add_intercept(self, X):
...
+
+        multiclass_meta = {"ovr": OneVsRestClassifier, "ovo": OneVsOneClassifier}[
+            multi_class
+        ]
+        return multiclass_meta(
+            _DemographicParityClassifer(*args, **kwargs), n_jobs=n_jobs
+        )
+
+
+@deprecated(
```
Comment (koaning) [inline sklego/linear_model.py]: the fairclassifier is depricated right, not the demographic parity one?

File: sklego/linear_model.py [hunk 2]
```diff
@@ -231,4 +189,150 @@ def _add_intercept(self, X):
...
+
+    @classmethod
+    def _get_param_names(cls):
+        return sorted(
+            super(_DemographicParityClassifer, cls)._get_param_names()
+            + _FairClassifier._get_param_names()
+        )
+
+
+class EqualOpportunityClassifier(BaseEstimator, LinearClassifierMixin):
```
Comment (koaning) [inline sklego/linear_model.py]: nice!

File: tests/test_estimators/test_demographic_parity.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from sklearn.linear_model import LogisticRegression
 
 from sklego.common import flatten
-from sklego.linear_model import FairClassifier
+from sklego.linear_model import DemographicParityClassifier
```
Comment (koaning) [inline tests/test_estimators/test_demographic_parity.py]: does it make sense to confirm that we have a deprecation warning of the deprecated model?

File: doc/fairness.ipynb [hunk 4]
```diff
@@ -115,11 +115,162 @@
    "source": [
```
Comment (koaning) [inline doc/fairness.ipynb]: image

the latex does not seem to render on the page? might be a github rendering issue but worthwhile to check.

File: sklego/linear_model.py (hunk 1 above)
Comment (MBrouns) [reply sklego/linear_model.py]: Thanks for catching that!

File: tests/test_estimators/test_demographic_parity.py (hunk 3 above)
Comment (MBrouns) [reply tests/test_estimators/test_demographic_parity.py]: Yes, will add that

File: doc/fairness.ipynb (hunk 4 above)
Comment (MBrouns) [reply doc/fairness.ipynb]: it seems to be a github issue as when I `make docs` it renders fine

**Comment by MBrouns** [comment]
So in contrast with the DemographicParityClassifier, this approach wasn't published so there is no formal benchmark run anywhere. It might be worth doing that before merging.

File: tests/test_estimators/test_demographic_parity.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    sensitive_cols = [0]
+    X_without_sens = np.delete(X, sensitive_cols, axis=1)
+    lr = LogisticRegression(penalty="none", solver="lbfgs")
+    fair = DemographicParityClassifier(
+        covariance_threshold=None, sensitive_cols=sensitive_cols, penalty="none"
+    )
+    try:
+        fair.fit(X, y)
+    except SolverError:
```
Comment (koaning) [inline tests/test_estimators/test_demographic_parity.py]: is there a good reason for this? something tells me that there should be no `SolverError` for the `random_xy_dataset_clf` datasets

**Comment by koaning** [comment]
@MBrouns whats the status on this? it is likely that I'll do a round of merges tomorrow, want this to go in?

File: tests/test_estimators/test_demographic_parity.py (hunk 5 above)
Comment (MBrouns) [reply tests/test_estimators/test_demographic_parity.py]: Absolutely right! I removed it
